frontend/src/pages/edit-driver-profile/components/edit-avatar-form/edit-avatar-form.tsx

```diff
@@ -0,0 +1,94 @@
+import { Button, ImageSelector } from '~/libs/components/components.js';
+import { ImgPath } from '~/libs/enums/enums.js';
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import {
+  useAppDispatch,
+  useAppForm,
+  useCallback,
+  useMemo,
+  useRef,
+} from '~/libs/hooks/hooks.js';
+import { useAuthUser } from '~/slices/auth/auth.js';
+import { uploadAvatar } from '~/slices/files/actions.js';
+
+import { type EditAvatarFormData } from './libs/types/types.js';
+import styles from './styles.module.scss';
+
+const AVATAR_SIZE = 200;
+
+const EditAvatarForm: React.FC = () => {
+  const { handleSubmit, setValue, register, formState } =
+    useAppForm<EditAvatarFormData>({
+      defaultValues: { avatar: undefined },
+    });
+  const { isDirty } = formState;
+
+  const user = useAuthUser();
+  const dispatch = useAppDispatch();
+
+  const inputReference = useRef<React.ElementRef<'input'>>(null);
+
+  const avatarUrl = useMemo(() => {
+    if (user && 'driver' in user) {
+      return user.driver.avatarUrl;
+    }
```

Comment:
```suggestion
    if (user.driver) {
      return user.driver.avatarUrl;
    }
```

Reply:
but actually edit avatar form should be displayed only for drivers, so maybe we don't need this check, and we can case user to necessary type ?
const user = useAuthUser() as NeededType;